### ISSUE TITLE
Remove git dependency?

### ISSUE THREAD

**Issue by iHiD** [skylightio/skylight-ruby#354]
[This line](https://github.com/skylightio/skylight-ruby/blob/4002121daf96db84d035c07b23f01a9fd18e7e3d/lib/skylight/util/deploy.rb#L108) requires the git executable to be installed. Our Dockerfiles are optimised to not have unnecessary things such as `git` installed, which means although the `.git` directory is there, this workflow fails for us.

Rather than relying on a binary, it might be a nicer option to read directly from the files that `git` itself reads from, which (I haven't checked, but I presume) are `.git/HEAD` and then whatever that points to (e.g. `.git/refs/heads/main` for our deployed app).

Here's an example of the config I'm trying out at Exercism for this: https://github.com/exercism/website/pull/1069/files

If we changed this upstream in this repo I could remove that messiness :)

**Comment by zvkemp** [comment]
The content inside your ERB tag looks generally ok (though you might want to `strip` the last File read). The sha should be nested under a `deploy` key:

```yml
deploy:
  git_sha: <%= ... %>
```

You can also set SKYLIGHT_DEPLOY_GIT_SHA in your env.

The stderr redirection already allows for `git` to be missing; I'm not sure if assuming the `.git` directory exists is necessarily _better_ than what we are already doing (e.g., it's commonly added to .dockerignore), but we will consider adding another Deploy fetcher for it.

**Comment by iHiD** [comment]
> though you might want to strip the last File read

Good point. Thanks. I'm actually setting it [here now](https://github.com/exercism/website/blob/main/config/environment.rb). Our security setup (very open source - no closed repos) means that we don't use environment variables and instead have everything powered internally by AWS's Secrets Manager, which is why this is a bit of a round-about way to do it in the Environment.rb file.

---

> I'm not sure if assuming the .git directory exists is necessarily better than what we are already doing (e.g., it's commonly added to .dockerignore)

If the `.git` directory is added to gitignore, the existing system breaks too though? As `git` will fail. 

Looking in the `.git` directory for the data, rather than relying on an executable, continues to work in the existing scenario, and also supports this extra scenario. It therefore adds functionality, removes a dependency and a system call, and has no downside that I can see, so I would argue it is better 🙂 

But it's your app! 🙂

**Comment by zvkemp** [comment]
Valid point. We'll put it on the roadmap.
